### PR TITLE
Add simplex/mixed mesh examples to simplex module page

### PR DESCRIPTION
**Pull request by peterrum**
This PR re-introduces `examples/doxygen` to be able to include simplex/mixed mesh examples in the simplex module page, while also be able to compile the code in CI.

### CODE REVIEW

File: examples/doxygen/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+PROJECT(doxygen_examples)
+
+DEAL_II_INITIALIZE_CACHED_VARIABLES()
+
+FOREACH(ccfile ${sources})
+  GET_FILENAME_COMPONENT(file ${ccfile} NAME_WE)
+  ADD_EXECUTABLE(${file} ${ccfile})
+  DEAL_II_SETUP_TARGET(${file})
+ENDFOREACH()
```
Comment (peterrum) [inline examples/doxygen/CMakeLists.txt:30]: This is not working, yet. For some reason, deal.II is not found. Any ideas?

File: doc/doxygen/headers/simplex.h [hunk 2]
```diff
@@ -85,4 +105,11 @@
  * documentation of the enumeration of the numbering of their vertices and
  * lines are given in the right columns.
  *
+ * @section simplex_reference_example_simplex Example: simplex mesh
+ *
+ * @include step_3_simplex.cc
+ *
+ * @section simplex_reference_example_mixed Example: mixed mesh
+ *
+ * @dontinclude step_3_mixed.cc
```
Comment (drwells) [inline doc/doxygen/headers/simplex.h:114]:
```suggestion
```

I'm not sure what your original intent here but it probably was not to include `step_3_simplex.cc` twice in this file. Should this be something else?

**Review by drwells — CHANGES_REQUESTED** [review]
I think its fine to add these examples in this way - it keeps them marked as experimental and outside of the official tutorial list.

**Comment by drwells** [comment]
I think I understand the CMake problem. You are running code during the initial CMake run that tries to find deal.II, which doesn't make sense: deal.II hasn't yet compiled. In contrast, the examples compiled via `make examples` are configured by first extracting their dependencies with text processing and then adding them via `DEAL_II_ADD_INSOURCE_TARGET` rather than using their individual `CMakeLists.txt`, which avoids this problem.

This is just for example compilation - the documentation is fine (and everything works if examples are disabled).

What do we gain by compiling the simplex examples in this way? Don't we end up compiling them anyway inside the CI run?

File: examples/CMakeLists.txt [hunk 3]
```diff
@@ -124,6 +124,33 @@ IF(DEAL_II_COMPONENT_EXAMPLES)
...
+            RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${DEAL_II_EXECUTABLE_RELDIR}"
+            )
+
+          IF(CMAKE_INSTALL_RPATH_USE_LINK_PATH)
+            SET_TARGET_PROPERTIES(${_name}.${_build_lowercase}
+              PROPERTIES
+              INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${DEAL_II_LIBRARY_RELDIR}"
+              )
+          ENDIF()
+
```
Comment (drwells) [inline examples/CMakeLists.txt:150]:
```suggestion
```

File: examples/CMakeLists.txt [hunk 4]
```diff
@@ -124,6 +124,33 @@ IF(DEAL_II_COMPONENT_EXAMPLES)
...
+        ENDFOREACH()
+
```
Comment (drwells) [inline examples/CMakeLists.txt:152]:
```suggestion
```

File: examples/CMakeLists.txt [hunk 5]
```diff
@@ -124,6 +124,33 @@ IF(DEAL_II_COMPONENT_EXAMPLES)
...
+            PROPERTIES
+            RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${DEAL_II_EXECUTABLE_RELDIR}"
+            )
+
+          IF(CMAKE_INSTALL_RPATH_USE_LINK_PATH)
+            SET_TARGET_PROPERTIES(${_name}.${_build_lowercase}
+              PROPERTIES
+              INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${DEAL_II_LIBRARY_RELDIR}"
+              )
+          ENDIF()
```
Comment (drwells) [inline examples/CMakeLists.txt:149]: If the examples are not installed (I don't see any calls to `INSTALL()` so I believe they are not) then this should not be set. Does this do something else that I don't understand?

**Review by drwells — CHANGES_REQUESTED** [review]
I have some minor CMake questions, otherwise 👍

**Comment by peterrum** [comment]
Done!

**Review by drwells — APPROVED** [review]

**Pull request merged**